### PR TITLE
[FIX] stock_picking_batch: add picks to batch

### PR DESCRIPTION
**Pull request by AhmedAmirK**
When adding list of transfers to batch from transfer tree view,
if transfers belong to two or more batches an error traceback pops.

To reproduce:

1-Create Transfer 1, go to Transfer list view and add it to batch under Action, choose
add to a new batch transfer and confirm.
2-Create Transfer 2 and do the same.
3-Now that we have two transfer belonging to two different batches,
Under Inventory->Operations->Transfers, select the two transfers
in the tree view and choose add to batch under Action, choose add to new batch
and confirm, error popsup

Bug originated from: #89815
Discovered while working on TaskId:2871679

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/stock_picking_batch/models/stock_picking_batch.py [hunk 1]
```diff
@@ -150,8 +150,9 @@ def create(self, vals):
 
     def write(self, vals):
         res = super().write(vals)
-        if not self.picking_ids and self.state == 'in_progress':
-            self.action_cancel()
+        for batch in self:
+            if not batch.picking_ids and batch.state == 'in_progress':
+                batch.action_cancel()
```
Comment (Whenrow) [inline addons/stock_picking_batch/models/stock_picking_batch.py:155]: small performance remark, you can call action_cancel once on all batches that need to be cancelled

Comment (ticodoo) [reply addons/stock_picking_batch/models/stock_picking_batch.py:155]: Ah I thought the same thing, but action_cancel has a `self.ensure_one()` in it. Any idea why it's there? And if it's safe to remove in stable?

Comment (AhmedAmirK) [reply addons/stock_picking_batch/models/stock_picking_batch.py:155]: it has self.ensure_one() 😢

Comment (Whenrow) [reply addons/stock_picking_batch/models/stock_picking_batch.py:155]: I'm surprised we still have action_* with ensure_one(). We should make it 'multi'

**Comment by Whenrow** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**